### PR TITLE
Sampling api

### PR DESCRIPTION
**Pull request by goedman**
This PR contains a proposed (simplified) sampling API as defined in StatsBase on top of MCMCChains.Chains objects as requested in #66.

It supports random sampling of draws in chains and sampling based on a Weights vector. This PR overloads StatsBase.sample() for that purpose and uses KernelDensity.kde() to compute a support range and associated densities. It is certainly possible to drop the KernelDensity dependency (although implicitly it is there through the density() plot).

### CODE REVIEW

**Comment by cpfiffer** [comment]
I'll merge this over in a while unless @trappmartin has any thoughts to throw in.

**Comment by trappmartin** [comment]
Great, I will have a look at it today.

File: src/sampling.jl [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import StatsBase: sample
+using Random, KernelDensity
+
+function sample(rng::AbstractRNG, chn::MCMCChains.AbstractChains, n;
+    replace=true, ordered=false)
```
Comment (yebai) [inline src/sampling.jl]: Some comments on the meaning of `replace` and `ordered` might be helpful here.

Comment (goedman) [reply src/sampling.jl]: True!

In this particular instance, ```sample()```,  StatsBase will step in:
```
? sample
```
will return:
```
help?> sample
search: sample sampler sample! samplepair Sampleable wsample wsample! nsamples

  sample([rng], a, [wv::AbstractWeights], n::Integer; replace=true, ordered=false)

  Select a random, optionally weighted sample of size n from an array a using a
  polyalgorithm. Sampling probabilities are proportional to the weights given in
  wv, if provided. replace dictates whether sampling is performed with replacement
  and order dictates whether an ordered sample, also called a sequential sample,
  should be taken.


  Optionally specify a random number generator rng as the first argument (defaults
  to Random.GLOBAL_RNG).
```

but in general, MCMCChains has enough how-to-use-it and features that it warrants it's own docs.

File: test/sampling_tests.jl [hunk 2]
```diff
@@ -0,0 +1,23 @@
+using Turing, MCMCChains, KernelDensity, StatsBase, Test, Statistics
+import StatsBase: sample
```
Comment (trappmartin) [inline test/sampling_tests.jl]: Why is this necessary?

File: test/sampling_tests.jl [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  sampler = HMC(500, 0.01, 5)
+  chn = sample(model, sampler, save_state=true);
+
+  chn_sample = sample(chn, 5)
+  @test range(chn_sample) == 1:1:5
+
+  c = kde(reshape(convert(Array{Float64}, chn[:s].value), 500))
+  chn_weighted_sample = sample(c.x, Weights(c.density), 100000)
+
+  @test 4.9 < mean(reshape(convert(Array{Float64}, chn[:s].value), 500)) < 5.1
```
Comment (trappmartin) [inline test/sampling_tests.jl]: Maybe replace with `@test mean(convert(Array{Float64}, chn[:s].value)) ≈ 5.0 atol=0.01`
